src/Aspire.Hosting.Azure.AppService/AzureAppServiceEnvironmentExtensions.cs

```diff
@@ -39,8 +39,9 @@ internal static IDistributedApplicationBuilder AddAzureAppServiceInfrastructureC
     /// </summary>
     /// <param name="builder">The distributed application builder.</param>
     /// <param name="name">The name of the resource.</param>
+    /// <param name="azureCloudName">The Azure cloud to deploy to. Defaults to <see cref="AzureCloudName.AzurePublic"/>.</param>
     /// <returns><see cref="IResourceBuilder{T}"/></returns>
-    public static IResourceBuilder<AzureAppServiceEnvironmentResource> AddAzureAppServiceEnvironment(this IDistributedApplicationBuilder builder, string name)
+    public static IResourceBuilder<AzureAppServiceEnvironmentResource> AddAzureAppServiceEnvironment(this IDistributedApplicationBuilder builder, string name, AzureCloudName azureCloudName = AzureCloudName.AzurePublic)
```

Comment:
Make this an extension method on the builder, not part of AddAzureAppServiceEnvironment. e.g. `WithCloudName`